### DISCUSSION TITLE
gjkRaycast of a capsule against convex taking a really long time to terminate when PhysX is built with clang and /fp:fast

### DISCUSSION THREAD

**Discussion by Zeblote** [NVIDIA-Omniverse/PhysX#123] [General]
Found this mysterious issue recently. I am building PhysX with clang as it makes much of my code quite a bit faster. However, there is one problem, and that's this happening sometimes:

![image](https://user-images.githubusercontent.com/6764788/226888687-93897ed0-ad48-49ae-b6ec-c7ca6f11bc58.png)

A sweep query taking almost a second to complete rather than the usual microseconds makes the game rather unplayable.

Neither the standard VC17 build nor a clang build with /fp:precise have this issue. What could be going on there? I'm not sure if this should be reported as a bug since using clang on windows is not the default, though /fp:fast is. Maybe the linux build would show the same problem since it is also using clang?

I've tried to put `#pragma float_control(precise, off)` in all the gjk functions but this had no effect at all, only setting /fp:precise for the entire library seems to fix the issue.

**Comment by preist-nvidia** [comment]
Hi @Zeblote - we are not aware of this issue on Linux - any chance you tried to repro with our defaults on Linux?

**Reply by Zeblote** [reply]
I will try in a bit, currently the linux build of the game is non-functional for unrelated reasons.

**Reply by Zeblote** [reply]
Finally got to check this again. The correct floating point arguments for clang-cl on windows are `/fp:fast /clang:-fhonor-infinities /clang:-fhonor-nans /clang:-fno-associative-math`. This solves the issue.

**Reply by preist-nvidia** [reply]
Thank you! We'll check and integrate!

**Reply by preist-nvidia** [reply]
Do you have a repro by any chance for the sweep lag?

**Reply by Zeblote** [reply]
Not off hand in a way I can share easily, it happens seemingly randomly on queries against our custom compound geometry, I would need to investigate a bit more and try to extract the specific shape that freezes it. Can potentially do that later this month.

**Reply by preist-nvidia** [reply]
Ok that would be very helpful to analyze the issue better, thank you.
